theme/layouts/_sidebar.ejs

```diff
@@ -0,0 +1,35 @@
+<%
+const query = site.queryPages;
+subPages = site.getSubPages;
+
+const navigation = [];
+for (const menuItem of site.menuItems) {
+  const page = query(({originalPathname}) => originalPathname === menuItem)[0];
+  navigation.push(subPages(page));
+}
+
+const [currentPage] = query(({pathname}) => page.pathname === pathname);
+let currentPathname = currentPage.originalPathname.split("/");
+pathnameList = [];
+for (let i=currentPathname.length-1; i >= 0; i--) {
+  pathnameList.push(currentPathname.join("/"));
+  currentPathname = currentPathname.slice(0,i);
+}
```

Comment:
Not sure if this is necessary if you will general `navigation` as suggested in the above comment, instead you will only need current page's `originalPathname` in order to not generate children of current page.